mathesar_ui/src/components/NameWithIcon.svelte

```diff
@@ -1,86 +1,55 @@
 <script lang="ts">
-  import { Icon, Spinner } from '@mathesar-component-library';
+  import { Icon, Spinner, Truncate } from '@mathesar-component-library';
   import type { IconProps } from '@mathesar-component-library/types';
 
   /** TODO: Update component and prop names */
   export let icon: IconProps | IconProps[];
   export let isLoading = false;
   /** When true, the icon will be rendered within a box */
   export let iconHasBox = false;
-  export let truncateName = true;
+  export let truncate = true;
 
   $: icons = Array.isArray(icon) ? icon : [icon];
 </script>
 
-<span class="name-with-icon" on:click class:boxed={iconHasBox}>
-  <span class="icon">
-    {#if isLoading}
-      <Spinner />
-    {:else}
-      {#each icons as icon}
-        <Icon
-          {...icon}
-          style={icons.length > 1 ? `width:${100 / icons.length}%` : undefined}
-        />
-      {/each}
-    {/if}
+<Truncate passthrough={!truncate}>
```

Comment:
> I don't like this pattern of encapsulating a component with another, without component specific classes present on the top-most parent element.

Hmmm. I think this topic warrants more discussion. If I'm understanding you correctly, we are already using this pattern fairly widely throughout the codebase. If we want to adopt the approach you're suggesting I think it would be good to agree on a code standard and document it. Would you be up for adding this to the front end meeting agenda so that we can chat about it more?